cpp/src/io/parquet/page_dict.cu

```diff
@@ -205,19 +203,14 @@ __global__ void __launch_bounds__(block_size, 1)
         row = frag_start_row + s->frag_dict[i + t];
         len = dtype_len;
         if (dtype == BYTE_ARRAY) {
-          const char *ptr = static_cast<const nvstrdesc_s *>(s->col.column_data_base)[row].ptr;
-          uint32_t count  = static_cast<uint32_t>(
-            static_cast<const nvstrdesc_s *>(s->col.column_data_base)[row].count);
-          len += count;
-          hash = nvstr_hash16(reinterpret_cast<const uint8_t *>(ptr), count);
+          auto str1 = s->col.leaf_column->element<string_view>(row);
+          len += str1.length();
+          hash = nvstr_hash16(reinterpret_cast<const uint8_t *>(str1.data()), str1.length());
           // Walk the list of rows with the same hash
           next_addr = &s->hashmap[hash];
           while ((next = atomicCAS(next_addr, 0, row + 1)) != 0) {
-            const char *ptr2 =
-              static_cast<const nvstrdesc_s *>(s->col.column_data_base)[next - 1].ptr;
-            uint32_t count2 =
-              static_cast<const nvstrdesc_s *>(s->col.column_data_base)[next - 1].count;
-            if (count2 == count && nvstr_is_equal(ptr, count, ptr2, count2)) {
+            auto str2 = s->col.leaf_column->element<string_view>(next - 1);
+            if (str1.length() == str2.length() && str1 == str2) {
```

Comment:
ping @davidwendt (apologies if this has been resolved offline)

Reply:
Fine with me.
https://github.com/rapidsai/cudf/blob/fc40c52a898181d3b4e019134e3ea0b32841515e/cpp/include/cudf/strings/string_view.cuh#L242-L245
But it would be more efficient to use `size_bytes()` instead of `length()`
```
__device__ inline bool string_view::operator==(const string_view& rhs) const
{
  return (size_bytes() == rhs.size_bytes()) && (compare(rhs) == 0);
}
```
```